### DISCUSSION TITLE
REST Query for Files under Top Level Folder

### DISCUSSION THREAD

**Discussion by st11x** [directus/directus#12492] [Archive]
Hi,

Say I have a folder structure like this:
A 
--> B
-----> C

How I I query all  files where the top level folder = 'A' ?

The deep parameter seems to do what I want, but I don't understand how to use it nor the examples given.

https://docs.directus.io/reference/query/#deep

The examples given doesn't refer to deep, and the API doesn't refer to the "_limit" that is shown in the examples. 

Thanks
  Matt

**Comment by rijkvanzanten** [comment]
> How I I query all files where the top level folder = 'A' ?

I think the most efficient way would be to use a regular `_in` filter on the /files endpoint:

```
GET /files?filter[folder][_in]=A,B,C
```

Alternatively, you could theoretically dive down the path from folder -> parent through

```
GET /files?filter[folder.parent.parent][_eq]=A
```

**Reply by st11x** [reply]
Thanks. The folders B and C are not fixed, there could be more levels. The only known folder is A. 

I also find that when requesting via the SEARCH Http method, it doesn't return the folder field. It does in the GET query, but I don't know how to construct a logical query with GET.

**Reply by rijkvanzanten** [reply]
> Thanks. The folders B and C are not fixed, there could be more levels. The only known folder is A.

There's currently no way to automatically recursively keep searching the parents until a given item is matched. If you know the maximum depth of folders, you could rely on that second approach though, as that wouldn't require you to know all folders ahead of time.

> I also find that when requesting via the SEARCH Http method, it doesn't return the folder field. It does in the GET query, but I don't know how to construct a logical query with GET.

Could you share what you tried? The SEARCH method+body uses the same systems as the GET request+query params, so all IO should be the same

**Comment by st11x** [comment]
Sorry I think I was mistaken about the SEARCH. One of my presets in PostMan had a different Auth token. That coupled with how the permissions were set up resulting in the mismatch. I see it working properly now. 

I suppose one workaround now is to apply a unique tag to everything under A.

**Reply by st11x** [reply]
I can't get the "in" filter to work on tags.

![image](https://user-images.githubusercontent.com/7398499/160942397-9ad6e313-0493-4687-b183-dabcc93cf7e4.png)

Two other images with empty tags came up only because I added some tags to them and then removed them. Those that never had tags would not show up at all. The one that I don't want to show is in the results.

**Reply by rijkvanzanten** [reply]
Directus currently isn't capable of searching in the stored value of JSON fields, see #7277 for the open feature request about that 👍🏻
